### ISSUE TITLE
Cannot restore from older backup version

### ISSUE THREAD

**Issue by comodoro** [duplicati/duplicati#3415]
- [x] I have searched open and closed issues for duplicates.

----------------------------------------

## Environment info

- **Duplicati version**: 2.0.3.11_canary_2018-09-05
- **Operating system**: Windows 10
- **Backend**: External HDD

## Description
Cannot restore files present in an older backup and deleted in newer backups

## Steps to reproduce

1. Make a backup
2. Delete some of the files you have just backed up and make another backup
3. Try to restore the deleted files from the UI or from the command line. 
4a. UI: Click Restore files. The deleted files do not appear in the file tree when correct backup in "Backup from" combo is selected
4b. Command line: 
```
Duplicati.CommandLine.exe restore file://E:\backup\all\ C:\Users\Vojta\Pictures --version=1
```
does not restore the files,

Variations that did not work:
```
Duplicati.CommandLine.exe restore file://E:\backup\all\ C:\Users\Vojta\Pictures\* --version=1
```
```
Duplicati.CommandLine.exe restore file://E:\backup\all\ C:\Users\Vojta\Pictures --version=1 --restore-path=d:\temp\restore
```
- **Actual result**:
UI: The files do not even show up for selection
Command line: 

```
Restore started at 10/8/2018 7:37:41 AM

Enter encryption passphrase: xxxxxx
  Listing remote folder ...
  Downloading file (65.41 MB) ...
  Downloading file (17.97 KB) ...
  Downloading file (17.95 KB) ...
...
[there is a lot of these - so perhaps the files still exist?]
...
  Downloading file (363.34 KB) ...
  Downloading file (17.97 KB) ...
  Downloading file (17.97 KB) ...
Checking remote backup ...
  Listing remote folder ...
Checking existing target files ...
  0 files need to be restored (0 bytes)
Verifying restored files ...
Restored 0 (0 bytes) files to original path
Duration of restore: 00:04:01
```
- **Expected result**:
The files should reappear

Is it likely that the files still exist, or are they gone? Is there standard way to browse the backups without Duplicati?

**Comment by comodoro** [comment]
Actually likely not since the `find` command did not find them

**Comment by Pectojin** [comment]
Do you have any retention policy on?

Also are you sure it's in backup version 1? What does "--all-versions = true" give you?

You can check how many backups should he available by looking for "dlist" files on your backend. There should be 1 for each backup. And technically you can inspect the file lists by decrypting those dlist manually, but this shouldn't be necessary.

**Comment by Pectojin** [comment]
Unless you've cleared files with `purge-broken-files`, then the only reason the files would silently fail to restore is if they don't match your pattern or if they don't exist in the backup (e.g. if you did two backups after they were removed pushing the files out of retention scope).

You can try browsing the restore from the Web UI from a machine with the server component installed. It can help you navigate the files to actually see if anything is there. When using restore from the web UI it will quickly build a local index before allowing you to browse, even if it's on a different machine.

**Comment by comodoro** [comment]
I have fortunately have found the files elsewhere. They are not in the old backup, although they definitely were present at the time and the backup is configured to include them. Contents of the directory instead match contents after its purge after Windows update several days ago. I realize this is a serious claim without proof, it is just a report of what happened.

> Unless you've cleared files with purge-broken-files, then the only reason the files would silently fail to restore is if they don't match your pattern or if they don't exist in the backup (e.g. if you did two backups after they were removed pushing the files out of retention scope).

Yes, please note that this has turned into an actual bug report. It states that the program does not behave as intended (as cited), at least in the canary version. However it is not up to me to decide whether to discard it.

> You can try browsing the restore from the Web UI from a machine with the server component installed. It can help you navigate the files to actually see if anything is there. When using restore from the web UI it will quickly build a local index before allowing you to browse, even if it's on a different machine.

Are there any advantages to using a different machine? Anyway local web UI shows that the files are not present, is there any reason to expect a different outcome?

**Comment by Pectojin** [comment]
No advantages using a different machine, but seeing as you're using the commandline I presumed the possibility of not having a GUI. 

For the purposes of reproduction, could you share exactly how you backed up these files?
I cannot replicate the issue with the provided steps.

**Comment by comodoro** [comment]
There is one important thing I failed to mention, I am sorry about that. Duplicati is not installed on the machine, it is from https://github.com/duplicati/duplicati/releases/download/v2.0.3.11-2.0.3.11_canary_2018-09-05/duplicati-2.0.3.11_canary_2018-09-05.zip. There did not appear to be any issues other than having to run the server or GUI manually.

I have checked the second backup log and found this error:
```
    Errors: [
        2018-10-06 16:08:42 +02 - [Error-Duplicati.Library.Main.Database.LocalBackupDatabase-CheckingErrorsForIssue1400]: Checking errors, related to #1400. Unexpected result count: 0, expected 1, hash: +6LzvJjeNubLmuiYCrz7vplbkae5RbvYouAoL0g6aUg=, size: 102400, blocksetid: 1317179, ix: 12, fullhash: MvxN03bnlVljvf/PV/l6KBf3YgOhXbPcrrGshvpwgd4=, fullsize: 1971184,
        2018-10-06 16:08:42 +02 - [Error-Duplicati.Library.Main.Database.LocalBackupDatabase-FoundIssue1400Error]: Found block with ID 1737364 and hash +6LzvJjeNubLmuiYCrz7vplbkae5RbvYouAoL0g6aUg= and size 93200
    ]
```

To your question, I ran the second from the GUI. I very likely have used command line for the first one, as there are no logs for it in the GUI (commands were created with the export to commandline GUI option, definitely not modified).

**Comment by ts678** [comment]
> I have checked the second backup log and found this error

As a general rule, errors abort backups, although more date information is needed in order to have that explain the situation.
Isssue1400 got fixed in [2.0.4.24_canary_2019-09-02](https://github.com/duplicati/duplicati/releases/tag/v2.0.4.24-2.0.4.24_canary_2019-09-02). I tried Steps to reproduce on latest Canary with 2 versions kept. All OK.
I doubt that we're going to get to the bottom of whatever happened in original issue. If nothing else is posted, this will close.
